### PR TITLE
Translation

### PR DESCRIPTION
**Pull request by marcusblake**
Added feature so that user can set a default language so that messages automatically appear in said language without having to translate each time.

### CODE REVIEW

**Review by zschreibs — COMMENTED** [review]
Overall looks good, you have to update the user-page.html to avoid a merge conflict (instead of all the addBasicLinkNavigation functions just use the loadNav function (I'm not 100% sure that's what its called) and then make sure to add the user settings page to that function.  Also I find that I'm not able to post messages on this branch, which is an issue i'm having on a few of them I don't know why though...

**Comment by zschreibs** [comment]
When I try and change the language in the user settings I get: 500 (Server Error) in the console, and the language does not change

**Comment by marcusblake** [comment]
I updated to fix the bug that you described. Added additional translating functionality which translates the About Me, Name, and Country. Also added small feature to highlight language when user selects a language. Also added functionality to translate text when visiting another user's page.

File: src/main/webapp/user-settings.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+<head>
+	<title>User Settings</title>
+	<meta charset="UTF-8">
+	<link rel="stylesheet" href="/css/main.css">
+	<script src="/js/navigation-loader.js"></script>
+	<script src="/js/user-settings.js"></script>
+</head>
+
+
+<body onload="addLoginOrLogoutLinkToNavigation(); addBasicLinkNavigation('stats.html', 'Stats Page'); addBasicLinkNavigation('feed.html', 'Public Feed');addBasicLinkNavigation('charts.html', 'Charts Page');addBasicLinkNavigation('map.html', 'Maps Page');addBasicLinkNavigation('user-map.html', 'User Map');
```
Comment (zschreibs) [inline src/main/webapp/user-settings.html]: change this whole thing to loadNavBar()

File: src/main/webapp/user-settings.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  	addBasicLinkNavigation('user-settings.html', 'User Settings');">
+
+<nav>
+    <ul id="navigation">
+      <li><a href="/">Home</a></li>
+    </ul>
```
Comment (zschreibs) [inline src/main/webapp/user-settings.html:22]: add
<li class="dropdown">
        <a href="javascript:void(0)" class="dropbtn">More</a>
        <ul class="dropdown-content" id="dropdown-content"></ul>
      </li>
under the Home li element so the dropdown menu still appears

**Review by zschreibs — COMMENTED** [review]
Looks good! Small edits but otherwise awesome!

**Review by zschreibs — APPROVED** [review]
Looks good!

**Pull request merged**